rpc/handlers.go

```diff
@@ -483,22 +483,35 @@ func (h *Handler) blockByID(id *BlockID) (*core.Block, *jsonrpc.Error) {
 	return block, nil
 }
 
-func (h *Handler) blockHeaderByID(id *BlockID) (*core.Header, error) {
+func (h *Handler) blockHeaderByID(id *BlockID) (*core.Header, *jsonrpc.Error) {
+	var header *core.Header
+	var err error
 	switch {
 	case id.Latest:
-		return h.bcReader.HeadsHeader()
+		header, err = h.bcReader.HeadsHeader()
 	case id.Hash != nil:
-		return h.bcReader.BlockHeaderByHash(id.Hash)
+		header, err = h.bcReader.BlockHeaderByHash(id.Hash)
 	case id.Pending:
-		pending, err := h.bcReader.Pending()
+		var pending blockchain.Pending
+		pending, err = h.bcReader.Pending()
 		if err != nil {
-			return nil, err
+			return nil, ErrInternal.CloneWithData(fmt.Errorf("get pending: %v", err))
```

Comment:
To make things simpler, we can assume Pending() can return db.ErrKeyNotFound as well.